R/compose.R

```diff
@@ -42,11 +42,37 @@ compose <- function(..., .rev = TRUE) {
     fns <- fns[-1]
   }
 
-  function(...) {
-    out <- first_fn(...)
+  body <- expr({
+    out <- !!fn_body(first_fn)
+
+    fns <- !!fns
     for (fn in fns) {
       out <- fn(out)
     }
+
     out
+  })
+
+  structure(
+    new_function(formals(first_fn), body, fn_env(first_fn)),
+    class = "purrr_composed_fn",
+    first_fn = first_fn,
+    fns = fns
+  )
+}
+
+print.purrr_composed_fn <- function(x, ...) {
+  cat("<composed>\n")
+
+  first <- attr(x, "first_fn")
+  cat("\n* Function 1:\n\n")
+  print(first, ...)
+
+  fns <- attr(x, "fns")
```

Comment:
This makes me think we should have a test for compiling compose with zero and one function.

Reply:
`compose()` with zero inputs now returns the identity such that `compose(fn, compose())` === `compose(fn)`